core/storage/trie/polkadot_trie/polkadot_trie_impl.hpp

```diff
@@ -81,6 +83,8 @@ namespace kagome::storage::trie {
     outcome::result<NodePtr> retrieveChild(const BranchNode &parent,
                                            uint8_t idx) override;
 
+    outcome::result<void> getValue(ValueAndHash &value) const override;
```

Comment:
Hm, it looks confusing because getters typically receive some immutable key and return the result. Can't this method just receive a hash (maybe as an rvalue) and return ValueAndHash? It would make it more intuitive to my mind.